include/zephyr/device.h

```diff
@@ -149,6 +149,8 @@ typedef int16_t device_handle_t;
 #define DEVICE_DT_NAME(node_id)                                                \
 	DT_PROP_OR(node_id, label, DT_NODE_FULL_NAME(node_id))
 
+#define DEVICE_DT_DEFER(node_id)                                               \
+	DT_PROP_OR(node_id, zephyr_deferred_init, false)
```

Comment:
`DT_PROP(node_id, zephyr_deferred_init)` (boolean prop, defaults to 0 if not defined)